### ISSUE TITLE
[react-apollo] Test is failing from Flow version v0.89.0

### ISSUE THREAD

**Issue by piglovesyou** [flow-typed/flow-typed#3161]
This is a build of my tiny PR and it failed. The cause doesn't look of mine. https://travis-ci.org/flow-typed/flow-typed/jobs/491182277

Also, when I develop locally using `react-apollo` types, at least `OperationComponent` is broken. Whatever you specify the generics value to `OperationComponent<_>`, it works just like specifying `any` to it.

```javascript
// @flow with version 0.89.0 or greater
const withUserData: OperationComponent<SomeQuery> = graphql(QueryGQL);

const MyComponent = withUserData(props => {

  // Expect error, but nothing is output.
  (props.data.someQuery.stringValue: number);
  return <div></div>;
});
```

**Comment by piglovesyou** [comment]
@wzrdzl @thymikee @jedwards1211 @budde377 I'd like to mention to some of common contributors/reviewers. I think "Not working" is some situation. It might be a problem of Flow itself but I'm not sure. So glad if I have comments from you.

**Comment by jedwards1211** [comment]
I'll have to look again, but last time I saw the CI failures it was a regression in Flow itself, not a problem with these type defs, and it didn't zeem like there would be an appropriate workaround, unfortunately.

**Comment by piglovesyou** [comment]
Thank you @jedwards1211 . I suspect it was Flow's regression too, as I looked through `react-apollo` type defs. Maybe I'll try creating an issue to Flow side, but right now I'm not too confident to find reproducing a minimum case.

**Comment by jedwards1211** [comment]
Yeah i had the same problem... guess you could try telling them you haven't been able to reduce to a minimum case and they should just debug on these type defs?

**Comment by piglovesyou** [comment]
Well I haven't yet but for not being like that I'll try a good one 👍 
Let me keep open this thread until we can decide which side the problem is.

**Comment by piglovesyou** [comment]
I'm afraid I couldn't... I ended up in switching TypeScript partially because of this problem.
